src/shared/cloudformation/cloudformation.ts

```diff
@@ -284,4 +304,209 @@ export namespace CloudFormation {
             resource.Properties.Handler === handlerName
         )
     }
+
+    /**
+     * Parameter/Ref helper functions
+     */
+
+    /**
+     * Validates whether or not a property is a valid ref.
+     * @param property Property to validate
+     */
+    function isRef(property: string | number | object | undefined): boolean {
+        return (
+            typeof property === 'object' && Object.keys(property).length === 1 && Object.keys(property).includes('Ref')
+        )
+    }
+
+    /**
+     * Gets the string value for a property in a template.
+     * If the value is a Ref to a parameter, returns the default value of the ref; this may be undefined.
+     * Also returns undefined if the property is undefined or a number.
+     * @param property Property value to check
+     * @param template Template object to parse through
+     */
+    export function getStringForProperty(
+        property: string | number | object | undefined,
+        template: Template
+    ): string | undefined {
+        if (validatePropertyType(property, 'string', template)) {
+            if (typeof property === 'string') {
+                return property
+            } else if (isRef(property)) {
+                try {
+                    const forcedProperty = property as Ref
+                    return getReffedString(forcedProperty, template)
+                } catch (err) {
+                    getLogger().debug(err)
+                }
+            }
+        }
+
+        return undefined
+    }
+
+    /**
+     * Gets the numeric value for a property in a template.
+     * If the value is a Ref to a parameter, returns the default value of the ref; this may be undefined.
+     * Also returns undefined if the property is undefined or a string.
+     * @param property Property value to check
+     * @param template Template object to parse through
+     */
+    export function getNumberForProperty(
```

Comment:
impl looks similar to `getStringForProperty`. Could put the implementation in a shared function like:

    function doGetProperty(..., propType: string, getFn: function)

Reply:
I think the issue was that the generic type wasn't working with the refs. I can give it a shot if you'd like, but the problem is that you can't get the type being used as type T, so you'd still need to have getReffedNumber/getReffedString separately.

The other alternative is just to return everything as string and have the caller parse to a number. The only thing that isn't represented purely as a string is a `Number` CFN param type, so it might be worth ditching this and only ever returning TS strings: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html#parameters-section-structure-properties . Would be interested to hear your thoughts on this in particular.